### ISSUE TITLE
OpenAPI document generation fails for a JsonArray as return type (Payara 5.194) /CUSTCOM-155

### ISSUE THREAD

**Issue by andbin** [payara/Payara#4420]
# Description #
----------

In a very basic webapp (for study purposes on MicroProfile 3.2) deployed on Payara 5.194 (tested on Full & Micro) I have a simple JAX-RS resource class like the following:

```
import javax.enterprise.context.ApplicationScoped;
import javax.json.Json;
import javax.json.JsonArray;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("/")
@ApplicationScoped
public class MainResource {

    @GET
    @Path("/greetings")
    @Produces(MediaType.APPLICATION_JSON)
    public JsonArray greetings() {
        return Json.createArrayBuilder().add("Hello").build();
    }
}
```

The `/openapi` endpoint **fails**, it generates an "empty"/minimal document like this:

```
paths: {}
components: {}
```

and (most **important**!), on console I see the following exception:

```
Caused by: java.lang.ClassCastException: java.lang.Class cannot be cast to java.lang.reflect.ParameterizedType
	at fish.payara.microprofile.openapi.impl.processor.ApplicationProcessor.createSchema(ApplicationProcessor.java:974)
	at fish.payara.microprofile.openapi.impl.processor.ApplicationProcessor.insertDefaultResponse(ApplicationProcessor.java:938)
	at fish.payara.microprofile.openapi.impl.processor.ApplicationProcessor.visitGET(ApplicationProcessor.java:183)
	at fish.payara.microprofile.openapi.impl.visitor.OpenApiWalker.processAnnotation(OpenApiWalker.java:272)
	at fish.payara.microprofile.openapi.impl.visitor.OpenApiWalker.processAnnotation(OpenApiWalker.java:256)
	at fish.payara.microprofile.openapi.impl.visitor.OpenApiWalker.processAnnotations(OpenApiWalker.java:217)
	at fish.payara.microprofile.openapi.impl.visitor.OpenApiWalker.accept(OpenApiWalker.java:166)
	at fish.payara.microprofile.openapi.impl.processor.ApplicationProcessor.process(ApplicationProcessor.java:158)
	at fish.payara.microprofile.openapi.impl.OpenApiService$OpenApiMapping.buildDocument(OpenApiService.java:291)
	... 58 more
```

If I return a populated JsonObject, there are no problems. Maybe the OpenAPI document generator doesn't like something that is into the JsonArray type.

Don't know if relevant, in the Maven `pom.xml` I have just only this dependency:

```
    <dependencies>
        <dependency>
            <groupId>org.eclipse.microprofile</groupId>
            <artifactId>microprofile</artifactId>
            <version>3.2</version>
            <type>pom</type>
            <scope>provided</scope>
        </dependency>
    </dependencies>
```

## Environment ##

- **Payara Version**: 5.194
- **Edition**: Full & Micro
- **JDK Version**: Oracle JDK: 8u212 / 8u231 / 11.0.5  (all 64-bit)
- **Operating System**: Windows (10)

**Comment by rdebusscher** [comment]
Hi Andrea,

I have created the internal issue `CUSTCOM-155` to fix problem.

Regards
Rudy

**Comment by bilabila** [comment]
Hi,
What about this issue?

**Comment by smillidge** [comment]
Fixed in 5.2020.3
